From: David Zeuthen <davidz@redhat.com>
Date: Thu, 12 Apr 2007 19:46:26 +0000 (-0400)
Subject: only start one hald-addon-keyboard addon
X-Git-Tag: HAL_0_2
X-Git-Url: http://gitweb.freedesktop.org/?p=hal.git;a=commitdiff;h=f22a37b7a7073b3398e9060e21bc1d12dc58783f

only start one hald-addon-keyboard addon

this patch fixes the problem that one device can have get started the
same hald-addon-keyboard addon multiple times. This work only together
with my patch for the contains_not FDI-file directive.

(similar to commit 5745da44f3bb43dafd9b53cbfcfc1218619a5263 on master)
---

--- a/fdi/policy/10osvendor/10-keyboard-policy.fdi
+++ b/fdi/policy/10osvendor/10-keyboard-policy.fdi
@@ -5,10 +5,14 @@
   <device>
     <match key="info.capabilities" contains="input">
       <match key="info.capabilities" contains="button">
-	<append key="info.addons" type="strlist">hald-addon-keyboard</append>
+        <match key="info.addons" contains_not="hald-addon-keyboard">
+	  <append key="info.addons" type="strlist">hald-addon-keyboard</append>
+        </match>
       </match>
       <match key="info.capabilities" contains="input.keyboard">
-	<append key="info.addons" type="strlist">hald-addon-keyboard</append>
+        <match key="info.addons" contains_not="hald-addon-keyboard">
+	  <append key="info.addons" type="strlist">hald-addon-keyboard</append>
+        </match>
         <match key="info.capabilities" contains_not="button">
 	  <append key="info.capabilities" type="strlist">button</append>
         </match>
